devicetypes/smartthings/zwave-mold-detector.src/zwave-mold-detector.groovy

```diff
@@ -0,0 +1,218 @@
+		}
+	}
+
+	log.debug "parse() result ${results.inspect()}"
+
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	log.debug "Wake Up Interval Report: ${cmd}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	log.debug "Event: ${cmd.event}, Notification type: ${cmd.notificationType}"
+	def result = []
+	def value
+	def description
+
+	if (cmd.notificationType == 0x10) {  // Mold Environment Detection
+		switch (cmd.event) {
+			case 0x00:
+				value = "good"
+				description = "Mold environment not detected"
+				break
+			case 0x02:
+				value = "unhealthy"
+				description = "Mold environment detected"
+				break
+			default:
+				log.debug "Not handled event type: ${cmd.event}"
+				break
+		}
+
+		result = createEvent(name: "moldHealthConcern", value: value, descriptionText: description, isStateChange: true, displayed: true)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [:]
+
+	switch (cmd.sensorType) {
+		case 0x01:
+			map.name = "temperature"
+			map.unit = temperatureScale
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmd.scale == 1 ? "F" : "C", cmd.precision)
+			map.displayed = true
+			map.isStateChange = true
+			break
+		case 0x05:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			map.displayed = true
+			map.isStateChange = true
+			break
+		case 0x0B:
+			map.name = "dewpoint"
+			map.unit = temperatureScale
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmd.scale == 1 ? "F" : "C", cmd.precision)
+			map.displayed = true
+			map.isStateChange = true
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	def result = createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+
+	cmds += secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05)) // humidity
+	cmds += secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01)) // temperature
+	cmds += secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x0B)) // dew point
+	cmds += secure(zwave.notificationV3.notificationGet(notificationType: 0x10)) // mold
+
+	if (!state.lastbatt || (now() - state.lastbatt) >= 10 * 60 * 60 * 1000) {
+		cmds += ["delay 1000",
+				 secure(zwave.batteryV1.batteryGet()),
+				 "delay 2000"
+		]
+	}
+
+	cmds += secure(zwave.wakeUpV2.wakeUpNoMoreInformation())
```

Comment:
I expect that you will not receive a response from all of these gets before the device goes back to sleep, especially if they're sent encapsulated.

Reply:
You might want to verify.

Reply:
hm, this is similar like in _zooz-4-in-1-sensor_, how do You think it would be better to poll these gets?

Reply:
I think if you sent the `noMoreInformation` in response to the report of the last poll that would be your best option.

So in the handler for the notificationReport.

Reply:
I've checked that and I didn't have a problem with responses from these four gets. I've created a method with these commands and added to notificationReport as well. What do You think?